File: benchmarks/python/dict_ops.py
import sys

n = int(sys.argv[1]) if len(sys.argv) > 1 else 10_000
d = {}

for i in range(n):
    key = "k" + str(i)
    d[key] = i

total = 0
for i in range(n):
    key = "k" + str(i)
    if key in d:
        total += d[key]

print(total)

File: benchmarks/node/dict_ops.js
"use strict";

const n = process.argv[2] !== undefined ? parseInt(process.argv[2], 10) : 10_000;
const d = Object.create(null);

for (let i = 0; i < n; i++) {
    d["k" + i] = i;
}

let total = 0;
for (let i = 0; i < n; i++) {
    const key = "k" + i;
    if (Object.prototype.hasOwnProperty.call(d, key)) {
        total += d[key];
    }
}

console.log(total);
